### PR TITLE
Publish helix tests in release

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/aspnet/AspNetCore/issues/8103

### CODE REVIEW

**Review by dougbu — COMMENTED** [review]
This certainly looks fine from an efficiency perspective -- release bits tend to run faster. But, `Debug.Assert(...)`s should be consistent. What is the root cause of flakiness related to our assertions?

**Comment by pakrym** [comment]
We can investigate the root cause later when we decide that testing debug builds is a good idea. For now, the goal is to make tests less flaky.

**Review by dougbu — APPROVED** [review]
:ok: I'm convinced 😺

**Review by HaoK — APPROVED** [review]

File: eng/helix/helix.proj [hunk 1]
```diff
@@ -21,6 +21,7 @@
     <EnableAzurePipelinesReporter>false</EnableAzurePipelinesReporter>
     <IsExternal>true</IsExternal>
     <Creator>aspnetcore</Creator>
+    <SkipInvalidConfigurations>true</SkipInvalidConfigurations>
```
Comment (HaoK) [inline eng/helix/helix.proj]: We should circle back with the helix sdk folks if we have to turn this back on, its this issue, we can reopen it to track if need https://github.com/dotnet/arcade/issues/1605

**Pull request merged**